src/Cli/dotnet/commands/dotnet-publish/Program.cs

```diff
@@ -55,6 +64,57 @@ public static PublishCommand FromParseResult(ParseResult parseResult, string msb
                 msbuildPath);
         }
 
+        public static string GetAutomaticConfigurationIfSpecified(
+            ParseResult parseResult,
+            string defaultedConfigurationProperty,
+            IEnumerable<string> slnOrProjectArgs,
+            Option<string> configOption)
+        {
+            ProjectInstance project = GetTargetedProject(parseResult, slnOrProjectArgs);
+            string releaseMode = "";
+
+            string releasePropertyFlag = project.GetPropertyValue(defaultedConfigurationProperty);
+            if (!string.IsNullOrEmpty(releasePropertyFlag))
+                releaseMode = releasePropertyFlag == "true" ? "Release" : "Debug";
+        
+            if (
+                !ConfigurationAlreadySpecified(parseResult, ref project, configOption) &&
+                !string.IsNullOrEmpty(releaseMode) &&
+                !slnOrProjectArgs.Any(arg => arg.Contains(defaultedConfigurationProperty))
+               )
+                return $"-property:configuration={releaseMode}";
+            else
+                return String.Empty;
+        }
+
+        private static ProjectInstance GetTargetedProject(ParseResult parseResult, IEnumerable<string> slnOrProjectArgs)
+        {
+            string potentialProject = slnOrProjectArgs
+                    .Where(arg => File.Exists(arg) &&
+                        LikeOperator.LikeString(arg, "*.*proj", VisualBasic.CompareMethod.Text))
+                    .ToList()
+                    .FirstOrDefault();
+
+            if (string.IsNullOrWhiteSpace(potentialProject))
+            {
+                try
+                {
+                    potentialProject = MsbuildProject.GetProjectFileFromDirectory(Directory.GetCurrentDirectory()).Name;
```

Comment:
dotnet run would ideally use this code because there is actually code duplication between this and the msbuildproject and msbuildprojectlocator. but we won't make that change here.